framework/utils/fmuExporter.py

```diff
@@ -0,0 +1,193 @@
+      cloudpickle.dump(self.model, pk, protocol=pickle.HIGHEST_PROTOCOL)
+    self.executeMethod = self._options.pop("executeMethod", None)
+    if self.executeMethod is None:
+      self.raiseAnError(IOError, "No executeMethod has been provided for FMU exporter!")
+    self.inputVars = self.model._getVariableList('input')
+    self.outVars =  self.model._getVariableList('output')
+    if not self.inputVars or not self.outVars:
+      self.raiseAnError(IOError, "Model {} is not exportable as FMU since no info about inputs/outputs are available!".format(self.model.name))
+    self.indexReturnDict = self._options.pop("indexReturnDict", None)
+    if self.indexReturnDict is None:
+      self.raiseAMessage("No indexReturnDict has been provided for FMU exporter! Default to 0!")
+      self.indexReturnDict = 0
+    self.frameworkDir = self._options.pop("frameworkDir")
+    if self.frameworkDir is None:
+      self.raiseAnError(IOError, "No frameworkDir has been provided for FMU exporter!")
+
+  def createModelHandler(self):
+    """
+      This create a temporary file that represents the model to export
+      @ In, None
+      @ Out, createModelHandler, str, the module for the FMU creation
+    """
+    classname = self.model.name.upper()
+    filename = self.model.name + ".pk"
+    # inputVars = ['{}'.format(el) for el in self.inputVars]
+    # outVars = ['{}'.format(el) for el in self.outVars]
+
+    return f"""
+import sys
+import pickle
+import os
+from pythonfmu.fmi2slave import Fmi2Type, Fmi2Slave, Fmi2Causality, Fmi2Variability, Integer, Real, Boolean, String
+
+class {classname}(Fmi2Slave):
+  #
+  #  RAVEN (raven.inl.gov) Model-based Python-driven simulator
+  #
+  author = "Andrea Alfonsi"
+  description = "RAVEN Model-based Python-driven simulator"
+
+  def __init__(self, **kwargs):
+    super().__init__(**kwargs)
+    self.inputVariables = {self.inputVars}
+    self.outputVariables = {self.outVars}
+    # set path to raven and the serialized model
+    self.raven_path = r"{self.frameworkDir}"
+    # model_path is by default the path to this model that is exported as FMU (serialized). It is stored in the resource folder
+    self.model_path = self.resources + "/" + "{filename}"
+    #os.path.sep
+    sys.path.append(self.model_path)
+    # this flag activates the initialization at the begin of the solve
+    self.initialized = False
+    # register raven_path variables if needed to be changed
+    self.register_variable(String("raven_path", causality=Fmi2Causality.parameter, variability=Fmi2Variability.tunable))
+    # register input variables
+    for var in self.inputVariables:
+      # set var
+      self.__dict__[var] = 0.0
+      self.register_variable(Real(var, causality=Fmi2Causality.input))
+    for var in self.outputVariables:
+      # set var
+      self.__dict__[var] = 0.0
+      self.register_variable(Real(var, causality=Fmi2Causality.output))
+
+  def setup_experiment(self, start_time: float):
+    self.start_time = start_time
```

Comment:
one more question here, do we want to follow raven python standard for this file? both camelBack and underscore are used in this file.

Reply:
At least some of the variable names are required by the the fmi standard.